packages/backend/src/open_payments/receiver/service.test.ts

```diff
@@ -0,0 +1,221 @@
+import { v4 as uuid } from 'uuid'
+
+import { ReceiverService } from './service'
+import { createTestApp, TestContainer } from '../../tests/app'
+import { Config } from '../../config/app'
+import { initIocContainer } from '../..'
+import { AppServices } from '../../app'
+import { createIncomingPayment } from '../../tests/incomingPayment'
+import { createPaymentPointer } from '../../tests/paymentPointer'
+import { truncateTables } from '../../tests/tableManager'
+import { ConnectionService } from '../connection/service'
+import { OpenPaymentsClient } from 'open-payments'
+import { PaymentPointerService } from '../payment_pointer/service'
+
+describe('Receiver Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let receiverService: ReceiverService
+  let openPaymentsClient: OpenPaymentsClient
+  let knex: Knex
+  let connectionService: ConnectionService
+  let paymentPointerService: PaymentPointerService
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    receiverService = await deps.use('receiverService')
+    openPaymentsClient = await deps.use('openPaymentsClient')
+    connectionService = await deps.use('connectionService')
+    paymentPointerService = await deps.use('paymentPointerService')
+    knex = await deps.use('knex')
+  })
+
+  afterEach(async (): Promise<void> => {
+    jest.useRealTimers()
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('get', () => {
+    describe('connections', () => {
+      const CONNECTION_PATH = 'connections'
+
+      test('resolves local connection', async () => {
+        const paymentPointer = await createPaymentPointer(deps, {
+          mockServerPort: Config.openPaymentsPort
+        })
+        const { connectionId } = await createIncomingPayment(deps, {
+          paymentPointerId: paymentPointer.id
+        })
+
+        const localUrl = `${Config.openPaymentsUrl}/${CONNECTION_PATH}/${connectionId}`
+
+        await expect(receiverService.get(localUrl)).resolves.toMatchObject({
+          assetCode: paymentPointer.asset.code,
+          assetScale: paymentPointer.asset.scale,
+          incomingAmount: undefined,
+          receivedAmount: undefined,
+          ilpAddress: expect.any(String),
+          sharedSecret: expect.any(Buffer),
+          expiresAt: undefined
+        })
+      })
+
+      test('resolves remote connection', async () => {
+        const paymentPointer = await createPaymentPointer(deps)
+        const incomingPayment = await createIncomingPayment(deps, {
+          paymentPointerId: paymentPointer.id
+        })
+
+        const remoteUrl = new URL(
+          `${paymentPointer.url}/${CONNECTION_PATH}/${incomingPayment.connectionId}`
+        )
+
+        const clientGetConnectionSpy = jest
+          .spyOn(openPaymentsClient.ilpStreamConnection, 'get')
```

Comment:
Could you spy (and not mock) this for the local tests, and check that it isn't called?

Reply:
Yup, good call